packages/docusaurus/src/server/plugins/init.ts

```diff
@@ -8,9 +8,36 @@
 import Module from 'module';
 import {join} from 'path';
 import importFresh from 'import-fresh';
-import {LoadContext, Plugin, PluginConfig} from '@docusaurus/types';
+import {
+  LoadContext,
+  Plugin,
+  PluginConfig,
+  ValidationSchema,
+} from '@docusaurus/types';
 import {CONFIG_FILE_NAME} from '../../constants';
 
+function validate<T>(schema: ValidationSchema<T>, options: Partial<T>) {
+  const {error, value} = schema.validate(options, {
+    convert: false,
+  });
+  if (error) {
+    throw error;
+  }
+  return value;
+}
+
+function validateAndStrip<T>(schema: ValidationSchema<T>, options: Partial<T>) {
+  const {error, value} = schema.validate(options, {
+    convert: false,
+    stripUnknown: true, // since the themeConfig is a shared object between plugins, it suppresses error and normalizes values relevant to the plugin
```

Comment:
What about applying stripUnknown  only to the top-level object?
https://hapi.dev/module/joi/api/?v=17.1.1#anyvalidatevalue-options

Reply:
I am not getting what are u saying. Do u want to remove keys from only on the themeConfig object?

Reply:
Hmmm the link was not correct:
https://hapi.dev/module/joi/api/?v=17.1.1#objectunknownallow


As the theme config is a global config, this plugin is only interested by some top level theme config attributes. But we should rather not allow arbitrary data on deeply nested attributes.

For example:

This should pass:

```js
const config = {
  themeConfig: {
    announcementBar: {
      id: "supportus",
      content: "blaa"
    },
    unknownAttribute: 42
  }
};
```

But this should fail, because we are sure `unknownAttribute` is not something we expect on `announcementBar` config:

```js
const config = {
  themeConfig: {
    announcementBar: {
      id: "supportus",
      content: "blaa",
      unknownAttribute: 42,
    },
  }
};
```


To make things simpler for the user, you could do this directly on the validate() fn:

```
const validate = (schema,value) => schema.unknown().validate(value,opts)
```


Also, if you don't strip top-level unknown theme attributes, you probably don't need to merge anymore

Reply:
but it will make it coupled to Joi since users may not be using JOi for validation

Reply:
no because if user wants to validate with something else, he'll not use the provided validate() function

Reply:
cool

Reply:
Done